### ISSUE TITLE
Provide an option to have a custom principal.getName() extractor

### ISSUE THREAD

**Issue by Plawn** [quarkiverse/quarkus-zanzibar#95]
If we don't wan to use the `principal.getName()` as the user identifier, we could provide a class like ?

```java
@ApplicationScoped
public class CustomZanzibarPrincipalExtractor implements ZanzibarPrincipalExtractor {
    public String extract(...) {

    }
}

```

**Comment by kdubb** [comment]
This seems like a worthwhile and relatively easily implemented idea.

**Comment by kdubb** [comment]
This has been released in 1.4.0... Note that `ZanzibarUserIdExtractor` was renamed to `UserIdExtractor` to be inline with other bean names.
